DomUtils.js

```diff
@@ -0,0 +1,150 @@
+        options[i].selected = (values.indexOf(options[i].value) > -1);
+      }
+    },
+
+    /**
+     * Verwijder de opties
+     * @param select {Object} select De 'select' lijst
+     */
+    resetSelectOptions: function (select) {
+      domConstruct.empty(select);
+    },
+
+    /**
+     * Voeg 'options' toe aan een 'select' lijst.
+     * @param {Object} select De 'select' lijst
+     * @param {Object} options Een object met de opties, formaat:
+     *   {
+     *     data: {array},
+     *     idProperty: {string},
+     *     labelProperty: {string}
+     *   }
+     */
+    addSelectOptions: function (select, options){
+      domConstruct.empty(select);
+      array.forEach(options.data, function (item) {
+        domConstruct.place(
+          '<option value="' + item[options.idProperty] + '">' + item[options.labelProperty] + '</option>', select);
+      });
+    },
+
+    /**
+     * Voeg 'options' inclusief placeholder toe aan een 'select' lijst.
+     * @param {Object} select De 'select' lijst
+     * @param {Object} options Een object met de opties, formaat:
+     *   {
+     *     data: {array},
+     *     idProperty: {string},
+     *     labelProperty: {string}
+     *     placeholder: undefined || {string}
+     *   }
+     */
+    addSelectOptionsWithPlaceholder: function (select, options){
+      var placeholder = options.placeholder || 'Kies een optie...';
+      domConstruct.empty(select);
+      domConstruct.place(
+        '<option value="-1" class="select-placeholder" selected>' + placeholder + '</option>', select);
+      array.forEach(options.data, function (item) {
+        domConstruct.place(
+          '<option value="' + item[options.idProperty] + '">' + item[options.labelProperty] + '</option>', select);
+      });
+    },
+
+    /**
+     * Verwijder de listitems
+     * @param select {Object} list De 'ul' of 'ol' lijst
+     */
+    resetList: function (list) {
+      domConstruct.empty(list);
+    },
+
+    /**
+     * Utility om arrays van objecten te mergen zonder dubbele entries (aan de hand van een uniek id)
+     * @param {array} array1 De eerste array
+     * @param {array} array2 De tweede array
+     * @param {String} idField De property die het uniek id bevat (default 'id');
+     */
+    mergeArrays: function (array1, array2, idField) {
+      idField = idField ? idField : 'id';
+      var nonUnique = array1.concat(array2);
+      var uniqueIds = {};
+      return array.filter(nonUnique, function(value) {
+        if (typeof value === 'object') {
+          if (!uniqueIds[value[idField]]) {
+            uniqueIds[value[idField]] = true;
+            return true;
+          }
+          return false;
+        } else {
+          if (!uniqueIds[value]) {
+            uniqueIds[value] = true;
+            return true;
+          }
+          return false;
+        }
+      });
+    },
+
+    makeArrayUnique: function (nonUniqueArray, idField) {
+      var unique = {};
```

Comment:
kan deze niet gebruitk worden in de mergearrays?
+ daar wordt een default waarde toegekend aan idField, hier niet